### PR TITLE
Crossreferences to standard library in mypy docs, part 6

### PR DESCRIPTION
**Pull request by hoefling**
Added references to the `more_types.rst`. This is the second-to-last part of splitting up the changes in #7624 into more readable PRs.

### CODE REVIEW

File: docs/source/more_types.rst [hunk 1]
```diff
@@ -608,12 +608,12 @@ value of type ``typing.Coroutine[Any, Any, T]``, which is a subtype of
...
 
 If you want to use coroutines in Python 3.4, which does not support
-the ``async def`` syntax, you can instead use the ``@asyncio.coroutine``
+the ``async def`` syntax, you can instead use the :py:func:`@asyncio.coroutine <asyncio.coroutine>`
 decorator to convert a generator into a coroutine.
 
 Note that we set the ``YieldType`` of the generator to be ``Any`` in the
 following example. This is because the exact yield type is an implementation
-detail of the coroutine runner (e.g. the ``asyncio`` event loop) and your
+detail of the coroutine runner (e.g. the :py:mod:`asyncio` :doc:`event loop <library/asyncio-eventloop>`) and your
```
Comment (hoefling) [inline docs/source/more_types.rst]: I think an additional link to the [event loop](https://docs.python.org/3/library/asyncio-eventloop.html) doc here would be helpful.

File: docs/source/more_types.rst [hunk 2]
```diff
@@ -884,33 +884,33 @@ Keys that aren't required are shown with a ``?`` in error messages:
...
-* ``d.copy()``
-* ``d.setdefault(key, default)``
-* ``d1.update(d2)``
-* ``d.pop(key[, default])`` (partial TypedDicts only)
-* ``del d[key]`` (partial TypedDicts only)
+* :py:meth:`d.copy() <dict.copy>`
+* :py:meth:`d.setdefault(key, default) <dict.setdefault>`
+* :py:meth:`d1.update(d2) <dict.update>`
+* :py:meth:`d.pop(key[, default]) <dict.pop>` (partial ``TypedDict``\s only)
+* ``del d[key]`` (partial ``TypedDict``\s only)
```
Comment (hoefling) [inline docs/source/more_types.rst]: Please advise whether referencing `dict` methods is useful here.

File: docs/source/more_types.rst (hunk 1 above)
Comment (JelleZijlstra) [reply docs/source/more_types.rst]: I think a single link is sufficient here. The asyncio event loop docs are not directly relevant here anyway.

Comment (hoefling) [reply docs/source/more_types.rst]: Agree, removed!

File: docs/source/more_types.rst (hunk 2 above)
Comment (gvanrossum) [reply docs/source/more_types.rst]: Yeah, I think it is useful.

**Pull request merged**